pyomo/common/fileutils.py

```diff
@@ -388,6 +390,34 @@ def find_executable(exename, cwd=True, include_PATH=True, pathlist=None):
                      pathlist=pathlist, allow_pathlist_deep_references=False)
 
 
+def import_file(path, clear_cache=False):
+    """
+    Import a module given the full path/filename of the file.
+    Replaces import_file from pyutilib (Pyomo 6.0.0).
+    
+    This function returns the module object that is created.
+    Parameters
+    ----------
+    path : str
+        Full path to .py file.
+    clear_cache: bool
+        Remove module if already loaded. The default is False.
+    """
+    path = os.path.expanduser(os.path.expandvars(path))
+    if not os.path.exists(path):
+        raise Exception('File does not exist. Check path.')
+    try:
+        module_dir, module_file = os.path.split(path)
+        module_name, module_ext = os.path.splitext(module_file)
+        if clear_cache and module_name in sys.modules:
+            del sys.modules[module_name]
+        spec = importlib.util.spec_from_file_location(module_name, path)
+        module = spec.loader.load_module()
+    except:
+        raise
```

Comment:
If we are going to leave this like this, then we should remove the try-except block entirely.  Personally, I think this is a good change, as suppressing the ImportError (or the SyntaxError) made debugging modules imported through import_file rather challenging.  However, we will need to go through all the places where import_file is called and (if necessary) replace the checks for `None` with handling of the raised exception.

Reply:
I did not find any checks for `None`. Other changes have been implemented.